content/en/docs/concepts/services-networking/topology-aware-hints.md

```diff
@@ -0,0 +1,127 @@
+---
+reviewers:
+- robscott
+title: Topology Aware Hints
+feature:
+  title: Topology Aware Hints
+  description: >
+    Enables topology aware routing with topology hints on EndpointSlices.
+
+content_type: concept
+weight: 10
+---
+
+
+<!-- overview -->
+
+{{< feature-state for_k8s_version="v1.21" state="alpha" >}}
+
+_Topology Aware Hints_ enable topology aware routing with topology hints
+included in EndpointSlices. This approach tries to keep traffic close to where
+it originated from. This can result in decreased costs and improved performance.
+
+<!-- body -->
+
+## Introduction
+
+Kubernetes clusters are increasingly deployed in multi-zone environments.
+_Topology Aware Hints_ adds new functionality that can help keep traffic within
+the zone it originated from. This concept is commonly referred to as "Topology
+Aware Routing". The "Hints" here refer to a key implementation detail. The
+EndpointSlice controller adds hints to EndpointSlices to allocate each endpoint
+to a zone. Those hints are then consumed by components like kube-proxy as they
+configure how requests are routed.
+
+## Using Topology Aware Hints
+
+This feature can be enabled for individual Services by setting the
+`service.kubernetes.io/topology-aware-hints` annotation to `auto`. This tells
+the EndpointSlice controller to set topology hints if it is deemed safe.
+Importantly, this does not guarantee that hints will always be set.
+
+## How it Works
+
+The functionality enabling this feature is split into two components: The
+EndpointSlice controller and Kube-Proxy. This provides a high level overview of
+how each component implements this feature.
+
+### EndpointSlice controller
+
+The EndpointSlice controller is responsible for setting hints on EndpointSlices
+when this feature is enabled. The controller allocates a proportional amount of
+endpoints to each zone. This proportion is based on the number of CPU cores that
+have been allocated to nodes running in that zone. For example, if one zone had
+2 CPU cores and another zone only had 1 CPU core, the controller would allocated
+twice as many endpoints to the zone with 2 CPU cores.
+
+### Kube-Proxy
+
+Kube-Proxy filters the endpoints it routes to based on the hints set by the
+EndpointSlice controller. Essentially, the controller allocates endpoints to
+specific zones, and kube-proxy is responsible for ensuring that it only routes
+to endpoints with hints for the zone it is in.
```

Comment:
Is it fair to say that the kube-proxy prefers same-zone endpoints and will try to send traffic to an endpoint in the same zone if the Service has added auto topology hints to the relevant EndpointSlice?

Reply:
Could we see an example EndpointSlice with a hint added (eg as an excerpt of a YAML dump)?

Reply:
Yep, that's a good way to phrase this. I can rework this to add an example.

Reply:
Added an EndpointSlice example and reworked this description. There's still enough nuance that it's difficult to explain well. Open to any ideas to improve the wording here.

Reply:
I'm happy that we don't need to iterate further on this part of the wording.